core/src/main/scala/pipeflow/system/iso8601/IntervalDuration.scala

```diff
@@ -0,0 +1,67 @@
+package pipeflow.system.iso8601
+
+import java.time.{Duration, Period}
+
+import scala.util.{Failure, Success, Try}
+
+
+object IntervalDuration {
+
+  class IntervalDurationParsingException(expression: String, cause: Throwable)
```

Comment:
I would prefer an exception free api :)

Reply:
I completely agree with that. In fact those exceptions are not thrown anywhere in the API, but basically used for the `Try` `Failure`s. See [this](https://github.com/chris-zen/pipeflow/blob/multi-project/core/src/main/scala/pipeflow/system/iso8601/IntervalDuration.scala#L37) and [this](https://github.com/chris-zen/pipeflow/blob/multi-project/core/src/main/scala/pipeflow/system/iso8601/IntervalDuration.scala#L46) for examples.
Would you suggest any alternative to `Try` in this case ?

Reply:
Yes we can use an Either with custom error messages

Reply:
I was avoiding the use of Either because of its limitations in Scala 2.10 and 2.11, but you just gave me the perfect excuse to introduce [cats](https://typelevel.org/cats), which I've been willing to learn since long ago.